src/main/java/com/getindata/connectors/http/internal/table/lookup/JavaNetHttpPollingClient.java

```diff
@@ -45,14 +69,25 @@ public Optional<RowData> pull(List<LookupArg> lookupArgs) {
 
     // TODO Add Retry Policy And configure TimeOut from properties
     private Optional<RowData> queryAndProcess(List<LookupArg> params) throws Exception {
-        URI uri = buildUri(params);
-        HttpRequest request =
-            HttpRequest.newBuilder().uri(uri).GET().timeout(Duration.ofMinutes(2)).build();
-        HttpResponse<String> response = httpClient.send(request, BodyHandlers.ofString());
 
+        HttpRequest request = buildHttpRequest(params);
+        HttpResponse<String> response = httpClient.send(request, BodyHandlers.ofString());
         return processHttpResponse(response);
     }
 
+    private HttpRequest buildHttpRequest(List<LookupArg> params) throws URISyntaxException {
+        URI uri = buildUri(params);
+        Builder requestBuilder = HttpRequest.newBuilder()
+            .uri(uri).GET()
+            .timeout(Duration.ofMinutes(2));
```

Comment:
Shouldn't the timeout be configurable (ofc with a provided default value)? And if we add timeout to Source, maybe we can add it to Sink too (we can leave `null` as a default value there)?

Reply:
Yes.
![image](https://user-images.githubusercontent.com/7932805/185366955-e4e4d659-a324-45e9-8cc2-c7ef22eeaf27.png)

BTW,
this timeout was there before this PR, its marked as new code, because I've extracted a method.

Reply:
Oh yes, you're right. Both the TODO and the fact that this code has been there already :)